### PR TITLE
feat: docs for payment form atom

### PR DESCRIPTION
**Pull request by Ryukemeister**
## What does this PR do?

- Added docs for the stripe payment form atom #17203 

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by Ryukemeister** [comment]
Only approved this after #17203 has been merged

**Review by supalarry — APPROVED** [review]

**Pull request merged**
